### PR TITLE
change setup-miniconda to use mamba instead

### PR DESCRIPTION
**Pull request by mkolopanis**
use mamba in github Action building
and circleci (this was _way_ too complicate to enable)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -22,30 +25,37 @@ jobs:
       - restore_cache:
           key: deps-{{ .Branch }}-{{ checksum "ci/pyuvdata_tests.yml" }}
       - run:
-          name: Install miniconda
-          command: |
-            wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O $HOME/miniconda.sh;
-            bash $HOME/miniconda.sh -b -p $HOME/miniconda
+          name: Instal Micromamba
```
Comment (bhazelton) [inline .circleci/config.yml:28]: Here and elsewhere in this file: "Instal" -> "Install"

File: .circleci/config.yml [hunk 2]
```diff
@@ -22,30 +25,37 @@ jobs:
...
+          command: wget -qO- https://micro.mamba.pm/api/micromamba/linux-64/latest | sudo tar -C / -xvj bin/micromamba
       - run:
           name: configure conda environment
           command: |
-            export PATH="$HOME/miniconda/bin:$PATH"
-            ./ci/install_conda.sh
+            micromamba info
+            # need these to add gxx and gcc to build novas and cython
+            micromamba create --name=${ENV_NAME}  python=$PYTHON gxx gcc -f ci/${ENV_NAME}.yml -yq
```
Comment (bhazelton) [inline .circleci/config.yml:35]: Does this mean we aren't using "install_conda.sh" in CircleCI any more?

File: ci/install_conda.sh [hunk 3]
```diff
@@ -1,24 +1,11 @@
...
-conda create --name=${ENV_NAME}  python=$PYTHON --quiet
-conda env update -n ${ENV_NAME} -f ci/${ENV_NAME}.yml
-source activate ${ENV_NAME}
-
-conda list -n ${ENV_NAME}
+micromamba info
+# need these to add gxx and gcc to build novas and cython
+micromamba create --name=${ENV_NAME}  python=$PYTHON gxx gcc -f ci/${ENV_NAME}.yml -yq
+micromamba activate ${ENV_NAME}
+micromamba list -n ${ENV_NAME}
```
Comment (bhazelton) [inline ci/install_conda.sh:8]: Are we using this anymore?

**Review by bhazelton — COMMENTED** [review]
Looks good. I found a typo and I have a question about the "install conda" script.

File: .circleci/config.yml (hunk 2 above)
Comment (mkolopanis) [reply .circleci/config.yml:35]: That's correct we're not. I can put some of this back in there if we'd like but ended up pulling it out while trying to debug.

**Comment by mkolopanis** [comment]
I've deleted the `install_conda.sh` script, if we end up bringing back azure, I can deal with re-implementing the logic there.

**Review by bhazelton — APPROVED** [review]
Looks good, thank you!

**Pull request merged**